daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -1695,9 +1695,15 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
           case erd: ElementRuntimeData => erd
           case _ => SDE("dfdlx:outputTypeCalcNextSibling can only be defined on an element")
         }
-        val resolver = erd.nextElementResolver
+        val resolver = erd.partialNextElementResolver
+        val followingERDs = resolver.allPossibleNextElements.dropWhile { _ == erd }
+        //
+        // FIXME: This is broken because it is looking outward and upward for enclosing contexts
+        // all the possible elements. This isn't possible anymore statically, since a group can be
+        // shared. You can do the immediate lexical siblings of an element only. Maybe that's sufficient?
```

Comment:
Add tests to cover this case where the sibling isn't properly situated relative to the one invoking this nextSibling function.

Reply:
There is a test for this. test_nextSibling_04